### PR TITLE
An 5383/nft sales updates

### PR DESCRIPTION
**Pull request by tarikceric**
create nft_sales_hadeswap_decoded and nft_sales_tensor_bid
- deprecate old hadeswap and update gold table

Decoded instructions return errors up to 2023-02-08 so use legacy model
- delete legacy records in gold via:
```
delete from SOLANA.nft.fact_nft_sales
where block_timestamp::date > '2023-02-08'
and program_id = 'hadeK9DLv9eA7ya5KCTqSvSvRZeJC3JgD5a9Y3CNbvu';
```

Runs:
- gold table run on m
`01:14:07  1 of 1 OK created sql incremental model nft.fact_nft_sales ..................... [SUCCESS 142359 in 13.71s]
`
- fr hadeswap and tensor_bid on 2xl
```
23:21:33  1 of 1 OK created sql incremental model silver.nft_sales_hadeswap_decoded ...... [SUCCESS 1 in 399.75s]
20:29:08  2 of 2 OK created sql incremental model silver.nft_sales_tensor_bid ............ [SUCCESS 1 in 10.30s]
```

- tests on hadeswap_decoded, tensor_bid and nft.fact_nft_sales
```
01:16:45  Finished running 63 data tests, 7 project hooks in 0 hours 1 minutes and 17.42 seconds (77.42s).
01:16:46  
01:16:46  Completed successfully
01:16:46  
01:16:46  Done. PASS=63 WARN=0 ERROR=0 SKIP=0 TOTAL=63
```

### CODE REVIEW

File: models/silver/nfts/silver__nft_sales_hadeswap_decoded.sql [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    AND A.buyer_escrow_vault = b.tx_from
+    AND A.index = b.index_1
+    GROUP BY
+        1, 2, 3, 4, 5, 6, 7, 8, 9
+)
+SELECT
+    block_timestamp,
+    block_id,
+    tx_id,
+    succeeded,
```
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_hadeswap_decoded.sql:126]: any reason to not surface `index`, `inner_index` in this model?

File: models/silver/nfts/silver__nft_sales_tensor_bid.sql [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        _inserted_timestamp
+    FROM 
+        silver.nft_sales_tensor_bid__intermediate_tmp
+)
+
+SELECT
+    block_timestamp,
+    block_id,
+    tx_id,
+    TRUE as succeeded,
```
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_tensor_bid.sql:71]: any reason to not surface `index`, `inner_index` in this model?

File: models/silver/nfts/silver__nft_sales_tensor_bid.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
+version: 2
+models:
+  - name: silver__nft_sales_tensor_bid
+    tests:
```
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_tensor_bid.yml:4]: Lets add a relationships test with `decoded_instructions_combined`

File: models/silver/nfts/silver__nft_sales_hadeswap_decoded.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
+version: 2
+models:
+  - name: silver__nft_sales_hadeswap_decoded
+    tests:
```
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_hadeswap_decoded.yml:4]: Lets add a relationships test with `decoded_instructions_combined`

File: models/silver/nfts/silver__nft_sales_tensor_bid.yml
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_tensor_bid.yml:1]: change `tests:` to `data_tests:`

File: models/silver/nfts/silver__nft_sales_hadeswap_decoded.yml
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_hadeswap_decoded.yml:1]: change `tests:` to `data_tests:`

**Review by desmond-hui — CHANGES_REQUESTED** [review]

File: models/silver/nfts/silver__nft_sales_tensor_bid.yml
Comment (tarikceric) [reply models/silver/nfts/silver__nft_sales_tensor_bid.yml:1]: updated

File: models/silver/nfts/silver__nft_sales_tensor_bid.yml (hunk 3 above)
Comment (tarikceric) [reply models/silver/nfts/silver__nft_sales_tensor_bid.yml:4]: added

File: models/silver/nfts/silver__nft_sales_hadeswap_decoded.sql (hunk 1 above)
Comment (tarikceric) [reply models/silver/nfts/silver__nft_sales_hadeswap_decoded.sql:126]: good point, added

**Review by desmond-hui — APPROVED** [review]

**Pull request merged**